### PR TITLE
Readable header

### PR DESCRIPTION
**Pull request by danielballan**
Here's an example of a Header for a run missing its RunStop. In the Python console, uids and PV strings are light grey, while important fields (Scan ID and the data keys) are green.

```
In [33]: print header
<Header>
Owner: benchmark_script
Project: None
Beamline ID: benchmark_b
Scan ID: 1903
Start Time: 2011-09-06 09:18:55.513500
Stop Time: Unknown
Exit Status: Unknown
Exit Reason: Unknown
..Event Descriptor (uid='aaac79f0-d61a-42a1-a2ce-5a80c883bcb0')
....linear_motor: PV:pv1
....scalar_detector: PV:pv2
....Tsam: PV:pv3
run_start_uid='601e56d4-aac7-4317-9cb3-904a5fa9dd9c'

In [34]: header
Out[34]: <Header run_start_uid='601e56d4-aac7-4317-9cb3-904a5fa9dd9c'>
```

### CODE REVIEW

**Comment by danielballan** [comment]
This PR relies on NSLS-II/metadatastore#113, which refactors `Document` into two classes, one that only handles the dottable-dict logic needed by `Header`, and another (the new `Document`) that handles the mongo-wrangling.

**Comment by danielballan** [comment]
P.S. This PR includes commits that finish the job of moving the nonscalar examples into MDS. As of this writing, they exist in both repos. The examples have to be revised in light of `Document` -> `Document.from_mongo`, so now seems like a good time to remove the duplicates.

**Comment by ericdill** [comment]
This problem that you're seeing when you run `python run_tests.py` with the `dottable` branch on metadatastore and this branch on dataportal is fixed by my PR.  I would rather not spend the time to try to fix a test that gets totally overhauled by my PR.  Is this ok with you?

```
======================================================================
FAIL: dataportal.tests.test_replay.test_replay_plotting1
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/edill/anaconda/envs/nikea2/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/edill/dev/python/dataportal/dataportal/tests/test_replay.py", line 149, in test_replay_plotting1
    if scalar_model.is_plotting]) == 1
AssertionError: 
```

**Comment by danielballan** [comment]
Absolutely

File: dataportal/broker/simple_broker.py [hunk 1]
```diff
@@ -279,63 +285,119 @@ def fill_event(event):
...
+                'beamline_id': 'beamline_id',
+                'owner': 'owner',
+                'group': 'group',
+                'project': 'project',
+                'run_start_id': 'id',
+                'run_start_uid': 'uid'}
+        for new_key, old_key in attrs.items():
+            header[new_key] = run_start[old_key]
+        if run_stop is not None:
+            attrs = {'stop_time': 'time',
```
Comment (tacaswell) [inline dataportal/broker/simple_broker.py:339]: not a huge fan of re-using variable names here.

File: dataportal/broker/simple_broker.py [hunk 2]
```diff
@@ -279,63 +285,119 @@ def fill_event(event):
...
+                        color='lightgrey')
+        s.write("run_start_uid='{0}'".format(self.run_start_uid),
+                'lightgrey')
+        if hasattr(self, 'run_stop_uid'):
+            s.write("run_stop_uid='{0}'".format(self.run_stop_uid),
+                    'lightgrey')
+        return s.readout()
+
+
+class Stream(object):
```
Comment (tacaswell) [inline dataportal/broker/simple_broker.py:386]: I would move this up to the utils module and give it a better name.

Comment (danielballan) [inline dataportal/broker/simple_broker.py:386]: I agree this is not a good place for it, but I need more time to think about where/how else we might use it (and, therefore, what it should be called).

File: dataportal/broker/simple_broker.py [hunk 3]
```diff
@@ -1,8 +1,11 @@
 from __future__ import print_function
+import sys
 import six  # noqa
 import copy
+from StringIO import StringIO
```
Comment (tacaswell) [inline dataportal/broker/simple_broker.py:5]: needs to be sixified.

File: dataportal/broker/simple_broker.py [hunk 4]
```diff
@@ -279,63 +285,119 @@ def fill_event(event):
...
-            error_msg = (
-                "{0} RunStop records (ids {1}) were found for the run with "
-                "run_start_id {2}".format(num, [rs.id for rs in run_stops],
-                                          run_start.id))
-            if verify_integrity:
-                raise IntegrityError(error_msg)
-            else:
-                warnings.warn(error_msg)
+    @classmethod
+    def from_run_start(cls, run_start):
```
Comment (ericdill) [inline dataportal/broker/simple_broker.py:292]: What's the rationale for making this a class method instead of just the `__init__`?

Comment (tacaswell) [inline dataportal/broker/simple_broker.py:292]: To make in more flexible.  There will probably be a case where we want to construct a `Header` object but won't be starting from a mongoengine objects.

This is a pretty common work around the fact that python does not have function overloading or multiple dispatch.  We might, for example, end up with a `Header.from_other_header` method and a `Header.from_v4_service` methods.

File: dataportal/broker/simple_broker.py (hunk 3 above)
Comment (ericdill) [inline dataportal/broker/simple_broker.py:5]: done in #114

File: dataportal/broker/simple_broker.py (hunk 4 above)
Comment (ericdill) [inline dataportal/broker/simple_broker.py:292]: solid point

**Comment by tacaswell** [comment]
included in #114

**Pull request closed without merging**